src/vcf/vcf_v42.ragel

```diff
@@ -194,12 +194,16 @@
         try {
           // Handle all columns and build record
           ParsePolicy::handle_body_line(*this);
+            try {
           // Check warnings (non-blocking errors but potential mistakes anyway, only makes sense if the last record parsed was correct)
           OptionalPolicy::optional_check_body_entry(*this, ParsingState::records->back());
-        } catch (ParsingError ex) {
-          ErrorPolicy::handle_body_section_error(*this, ex.what());
-        } catch (ParsingWarning ex) {
-          ErrorPolicy::handle_body_section_warning(*this, ex.what());
+            } catch (MetaSectionError &warn) {
+                ErrorPolicy::handle_warning(*this, warn);
+            } catch (BodySectionError &warn) {
+                ErrorPolicy::handle_warning(*this, warn);
+            }
+        } catch (BodySectionError &error) {
+            ErrorPolicy::handle_error(*this, error);
```

Comment:
There could be 2 try-catch blocks instead of 1 to avoid nesting, since ErrorPolicy::handle_error is only going to apply to ParsePolicy::handle_body_line anyway.